### PR TITLE
[Feature] Enhance the DelegationRssShuffleManager to support hdfs conf file and reset the shuffle manager name

### PR DESCRIPTION
**Pull request by duanmeng**
### What changes were proposed in this pull request?
1. Enhance the candidates checker to support hdfs
2. Reset the shuffle manager name on the fly

### Why are the changes needed?
It is more feasible to support both local and hdfs and we need to reset shuffle manager name the adaptive ESS when fallback to SortShuffleManager, in which the only the sort shuffle manager is supported.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
UT & system test

### CODE REVIEW

File: coordinator/pom.xml [hunk 1]
```diff
@@ -70,6 +70,24 @@
       <scope>test</scope>
       <type>test-jar</type>
     </dependency>
+    <dependency>
+      <groupId>com.tencent.rss</groupId>
+      <artifactId>shuffle-storage</artifactId>
```
Comment (colinmjj) [inline coordinator/pom.xml:75]: coordinator shouldn't depend on shuffle-storage module

File: client-spark/spark2/src/main/java/org/apache/spark/shuffle/DelegationRssShuffleManager.java [hunk 2]
```diff
@@ -71,16 +71,18 @@ private ShuffleManager createShuffleManagerInDriver() throws RssException {
       try {
         shuffleManager = new RssShuffleManager(sparkConf, true);
         sparkConf.set(RssClientConfig.RSS_ENABLED, "true");
+        sparkConf.set("spark.shuffle.manager", Constants.RSS_SHUFFLE_MANAGER_NAME);
```
[the remark this answers is not in the document]
Comment (duanmeng) [reply client-spark/spark2/src/main/java/org/apache/spark/shuffle/DelegationRssShuffleManager.java:74]: sure

File: coordinator/pom.xml (hunk 1 above)
Comment (duanmeng) [reply coordinator/pom.xml:75]: our hdfs related utils are in this module and  we need to support hdfs candidates file.

**Review by colinmjj — APPROVED** [review]
LGTM, depend on UT

**Pull request merged**